db_builder/Source/filedownloader.cpp

```diff
@@ -0,0 +1,118 @@
+#include <iostream>
+#include <stdlib.h>
+#include <fstream>
+#include <stdio.h>
+#include <string>
+#include <filesystem>
+#include <curl/curl.h>
+
+#include "filedownloader.h"
+
+using namespace fd;
+namespace fs = std::filesystem;
+
+FileDownloader::FileDownloader()
+{ 
+    curl = curl_easy_init();
+}
+
+bool FileDownloader::isFileDownloaded(std::string fileName_)
+{  
+    std::string c = directoryPath + "/" + fileName_;
+    std::ifstream ifexfile(c.c_str()); 
+    if(ifexfile)
+    {
+        std::string msg = fileName_ + " already exists!";
+        std::cout<<msg<<std::endl;
+        
+        return true;        
+    }
+    else 
+    {
+        return false;
+    }
+}
+
+void FileDownloader::addDirectoryPath(std::string directoryPath_)
+{
+    directoryPath = directoryPath_;
+}
+
+void FileDownloader::createDirectory(std::string directoryPath_)
+{
+    directoryPath = directoryPath_;
+    
+    if(fs::exists(directoryPath))
+    {
+        //printf("Directory already exists! \n");
+    }
+    else 
+    {
+        fs::create_directory(directoryPath);
+    }
+}
+
+void FileDownloader::addLinks(std::string links_[], std::string fileNames_[], int numberOfLinks)
+{
+    for(int i = 0; i < numberOfLinks; i++)
+    {
+        std::string l = links_[i];
+        links.push_back(l);
+        
+        std::string n = fileNames_[i];
+        fileNames.push_back(n);
+    }
+}
+
+void FileDownloader::download(bool checkIfExists)
```

Comment:
ten bool mi tutaj nie pasuje. Rozumiem, że to jest jakas logiczna flaga, która ma sterować zachowaniem funckji ? 

To raczej powinno być wrzucone do osobnej funkcji, w której byłoby sprawdzanie tego warunku. Bo ten fragment kodu, nie ma nic wspolnego z download

Reply:
dodałem funkcję do sterowania download()